src/app/services/user-location.service.ts

```diff
@@ -25,6 +27,8 @@ export class UserLocationService implements OnInit{
         this.currLat = pos.lat;
         this.currLng = pos.lng;
         this.currLocation = pos.lat + " " + pos.lng;
+        localStorage.setItem("currLat", this.currLat.toString());
```

Comment:
Do not store anything in the localStorage. This is a service. The other components should already have access to this value.